docs/tutorials/rerank/rerank_pipeline_with_Cohere_Rerank_model.md

```diff
@@ -0,0 +1,305 @@
+
+Note: Replace the placeholders that start with `your_` with your own values.
+
+# Steps
+
+## 1. Create Cohere Rerank model
+
+Create a connector:
+```
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "cohere-rerank",
+    "description": "The connector to Cohere reanker model",
+    "version": "1",
+    "protocol": "http",
+    "credential": {
+        "cohere_key": "your_cohere_api_key"
+    },
+    "parameters": {
+        "model": "rerank-english-v2.0"
+    },
+    "actions": [
+        {
+            "action_type": "predict",
+            "method": "POST",
+            "url": "https://api.cohere.ai/v1/rerank",
+            "headers": {
+                "Authorization": "Bearer ${credential.cohere_key}"
+            },
+            "request_body": "{ \"documents\": ${parameters.documents}, \"query\": \"${parameters.query}\", \"model\": \"${parameters.model}\", \"top_n\": ${parameters.top_n} }",
+            "pre_process_function": "connector.pre_process.cohere.rerank",
+            "post_process_function": "connector.post_process.cohere.rerank"
+        }
+    ]
+}
+```
+Use the connector ID from the response to create a model:
+```
+POST /_plugins/_ml/models/_register?deploy=true
+{
+    "name": "cohere rerank model",
+    "function_name": "remote",
+    "description": "test rerank model",
+    "connector_id": "your_connector_id"
+}
+```
+Note the model ID in the response; you will use it in the following steps.
+
+Test the model using the Predict API:
+```
+POST _plugins/_ml/models/your_model_id/_predict
+{
+  "parameters": {
+    "query": "What is the capital of the United States?",
+    "documents": [
+      "Carson City is the capital city of the American state of Nevada.",
+      "The Commonwealth of the Northern Mariana Islands is a group of islands in the Pacific Ocean. Its capital is Saipan.",
+      "Washington, D.C. (also known as simply Washington or D.C., and officially as the District of Columbia) is the capital of the United States. It is a federal district.",
+      "Capital punishment (the death penalty) has existed in the United States since beforethe United States was a country. As of 2017, capital punishment is legal in 30 of the 50 states."
+    ],
+    "top_n": 4
+  }
+}
+```
+Sample response:
+
+Explanation of the response:
+1. The response contains 4 `similarity` outputs. For each `similarity` output, the `data` array contains a relevance score between each document and the query.
+2. The `similarity` outputs are provided in the order of the input documents; the first result of similarity pertains to the first document.
+This differs from the default output of the Cohere Rerank model, which orders documents by relevance scores.
+The document order is changed in the `connector.post_process.cohere.rerank` post-processing function in order to make the output compatible with a reranking pipeline.
+```
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "similarity",
```

Comment:
I think it's worth mentioning in the explanation that even though the `inference_results.output.data_type` says float32, its not representative of the embeddings defined in the connector (`int8`)